DMCompiler/SpacemanDmm/AstConverter.cs

```diff
@@ -0,0 +1,853 @@
+                    return new DMASTCall(ConvertCallArgs(left), ConvertCallArgs(right));
+
+                case Term.Expr(var expr):
+                    return ConvertExpression(expr);
+
+                case Term.Float(var f):
+                    return new DMASTConstantFloat(f);
+
+                case Term.Ident(var identName):
+                    if (identName == ".")
+                        return new DMASTCallableSelf();
+
+                    return new DMASTIdentifier(identName);
+
+                case Term.Input input:
+                    return new DMASTInput(
+                        ConvertCallArgs(input.Args),
+                        (DMValueType?)input.InputType ?? DMValueType.Text,
+                        ConvertExpression(input.InList));
+
+                case Term.Int(var i):
+                    return new DMASTConstantInteger(i);
+
+                case Term.InterpString interpString:
+                    // TODO: Impl
+                    return new DMASTConstantString(interpString.Start);
+
+                case Term.List(var values):
+                    return new DMASTList(ConvertCallArgs(values));
+
+                case Term.Locate locate:
+                    if (locate.Args.Length == 3)
+                    {
+                        return new DMASTLocateCoordinates(
+                            ConvertExpression(locate.Args[0]),
+                            ConvertExpression(locate.Args[1]),
+                            ConvertExpression(locate.Args[2]));
+                    }
+
+                    var container = locate.InList;
+                    Expression? locateType = null;
+
+                    if (locate.Args.Length == 2)
+                    {
+                        locateType = locate.Args[0];
+                        container = locate.Args[1];
+                    }
+                    else if (locate.Args.Length == 1)
+                    {
+                        locateType = locate.Args[0];
+                    }
+
+                    return new DMASTLocate(ConvertExpression(locateType), ConvertExpression(container));
+
+                case Term.New(var newType, var args):
+                    var argsConverted = args == null ? Array.Empty<DMASTCallParameter>() : ConvertCallArgs(args);
+
+                    switch (newType)
+                    {
+                        case NewType.Implicit:
+                            return new DMASTNewInferred(argsConverted);
+                        case NewType.MiniExpr miniExpr:
+                            var dereferences = miniExpr.Fields.Select(f =>
+                            {
+                                var (conditional, type) = ConvertPropertyAccessKind(f.Kind);
+                                return new DMASTDereference.Dereference(type, conditional, f.Ident);
+                            }).ToArray();
+                            return new DMASTDereference(new DMASTIdentifier(miniExpr.Ident), dereferences);
```

Comment:
```suggestion
                            var dereference = new DMASTDereference(new DMASTIdentifier(miniExpr.Ident), dereferences);

                            return new DMASTNewDereference(dereference, argsConverted);
```

This fixes an issue with TestGame that was preventing the player mob from being created.